scripts/quests/sandoria/Blackmail.lua

```diff
@@ -0,0 +1,125 @@
+local quest = Quest:new(xi.questLog.SANDORIA, xi.quest.id.sandoria.BLACKMAIL)
+
+quest.reward =
+{
+    fame     = 30,
+    fameArea = xi.fameArea.SANDORIA,
+    gil      = 900,
+}
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_AVAILABLE and
+            player:getFameLevel(xi.fameArea.SANDORIA) >= 3 and
+            player:getRank(player:getNation()) >= 3
+        end,
+
+        [xi.zone.NORTHERN_SAN_DORIA] =
+        {
+            ['Dauperiat']  = quest:progressEvent(643),
+
+            onEventFinish =
+            {
+                [643] = function(player, csid, option, npc)
+                    quest:begin(player)
+                    npcUtil.giveKeyItem(player, xi.ki.SUSPICIOUS_ENVELOPE)
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status ~= xi.questStatus.QUEST_AVAILABLE
+        end,
+
+        [xi.zone.NORTHERN_SAN_DORIA] =
+        {
+            ['Dauperiat'] =
+            {
+                onTrade = function(player, npc, trade)
+                    if
+                        npcUtil.tradeHasExactly(trade, { { xi.item.COPY_OF_THE_CASTLE_FLOOR_PLANS, 1 } }) and
+                        quest:getVar(player, 'Prog') == 2 or
+                        quest:getVar(player, 'Repeat') == 1
+                    then
+                        return quest:progressEvent(648, 0, xi.item.COPY_OF_THE_CASTLE_FLOOR_PLANS)
+                    end
+                end,
+
+                onTrigger = function(player, npc)
+                    local questProgress = quest:getVar(player, 'Prog')
+                    local questRepeat   = quest:getVar(player, 'Repeat')
+
+                    if player:hasKeyItem(xi.ki.SUSPICIOUS_ENVELOPE) then
+                        return quest:event(645)
+                    elseif questProgress == 1 then
+                        return quest:progressEvent(646, 0, xi.item.COPY_OF_THE_CASTLE_FLOOR_PLANS)
+                    elseif questProgress == 2 then
+                        return quest:event(647, 0, xi.item.COPY_OF_THE_CASTLE_FLOOR_PLANS)
+                    elseif questRepeat == 0 then
+                        return quest:progressEvent(650, 0, xi.item.COPY_OF_THE_CASTLE_FLOOR_PLANS)
+                    elseif questRepeat == 1 then
+                        return quest:event(647, 0, xi.item.COPY_OF_THE_CASTLE_FLOOR_PLANS)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [646] = function(player, csid, option, npc)
+                    if option == 1 then
+                        quest:setVar(player, 'Prog', 2)
+                    end
+                end,
+
+                [648] = function(player, csid, option, npc)
+                    if player:hasCompletedQuest(quest.areaId, quest.questId) then
+                        quest.fame = 5
```

Comment:
This is incorrect.  `fame` attached to the quest object does nothing.

Reply:
Also, lua tables are passed by reference in most places.  If your intent is to modify quest.rewards, it will need to be tested with a separate character as well following the modification to ensure it is not permanently changed (my bet is it will be)

Reply:
Good point. Ill just add the addFame() instead.